src/AutoCompleteTextField.js

```diff
@@ -307,11 +308,11 @@ class AutocompleteTextField extends React.Component {
           break;
         case KEY_UP:
           event.preventDefault();
-          this.setState({ selection: ((options.length + selection) - 1) % options.length });
+          this.setState({ selection: ((optionsCount + selection) - 1) % optionsCount });
```

Comment:
I know it's __currently__ an unreachable state for `optionsCount` to be zero as `helperVisible` would be false but it still makes me a bit nervous.  While it's not directly related to your change can we still update logic here and below to
```
case KEY_UP:
          event.preventDefault();
          if (optionsCount > 0) {
              this.setState({ selection: Math.max(0, optionsCount + selection - 1) % optionsCount });
         }
         break;
...
          }
```